File: src/main/java/com/azilzor/chatterchain/security/OAuth2UserInfoExtractor.java
package com.azilzor.chatterchain.security;

import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
import org.springframework.security.oauth2.core.user.OAuth2User;

public interface OAuth2UserInfoExtractor {

  CustomUserDetails extractUserInfo(OAuth2User oAuth2User);

  Boolean accepts(OAuth2UserRequest userRequest);

  default String retrieveAttribute(String attribute, OAuth2User oAuth2User) {
    Object atribute = oAuth2User.getAttributes().get(attribute);
    return atribute == null ? "" : atribute.toString();
  }

}

File: src/main/kotlin/com/azilzor/chatterchain/security/OAuth2UserInfoExtractor.kt
package com.azilzor.chatterchain.security

import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest
import org.springframework.security.oauth2.core.user.OAuth2User

interface OAuth2UserInfoExtractor {
    fun extractUserInfo(oAuth2User: OAuth2User): CustomUserDetails

    fun accepts(userRequest: OAuth2UserRequest): Boolean

    fun retrieveAttribute(attribute: String?, oAuth2User: OAuth2User): String {
        val attributes = oAuth2User.attributes[attribute]
        return attributes?.toString() ?: ""
    }
}
